code/modules/organs/subtypes/parasite.dm

```diff
@@ -518,3 +518,112 @@
+	subtle = 1
+
+	drug_resistance = 1
+
+	max_stage = 4
+	stage_interval = 300 //~5 minutes minutes/stage
+
+	origin_tech = list(TECH_BIO = 3)
+
+	egg = /singleton/reagent/toxin/malignant_tumour_cells
+
+/obj/item/organ/internal/parasite/malignant_tumour/process()
+	..()
+	if (!owner)
+		return
+	if(REAGENT_VOLUME(owner.reagents, /singleton/reagent/ryetalyn))
+		recession = 5
+	if(prob(5))
+		owner.adjustNutritionLoss(5)
+		if(prob(5))
+			owner.adjustHalLoss((10*stage)/2)
+	if(stage >= 2)  //after ~5 minutes
+		switch(parent_organ)
+			if(BP_HEAD)
+				owner.confused = min(owner.confused + (stage*2), 10)
+				owner.slurring = min(owner.slurring + (stage*2), 50)
+				if(prob(5))
+					to_chat(owner, SPAN_WARNING(pick("You struggle to remember the last several moments.", "You smell something funny.", "You taste something funny.", "You can't figure out how to properly string words together.", "What was I just doing?")))
+					owner.emote("drool")
+			if(BP_CHEST)
+				if(prob(3))
+					to_chat(owner, SPAN_WARNING(pick("You feel a tightness in your chest.", "You feel a little lightheaded.", "You need to take larger breaths than normal.")))
+					owner.emote(pick("cough", "gasp"))
+			if(BP_GROIN)
+				if(prob(3))
+					to_chat(owner, SPAN_WARNING(pick("You feel a sharp pain in your gut.", "You feel a little bloated.")))
+					owner.visible_message("<b>[owner]</b> winces slightly.")
+					owner.delayed_vomit()
+			else
+				if(prob(stage*2))
+					to_chat(owner, SPAN_WARNING("You feel very lethargic."))
+	if(stage >= 3) //after ~10 minutes
+		for(var/obj/item/organ/internal/O in owner.internal_organs)
+			if(O.parent_organ == parent_organ)
+				if(BP_IS_ROBOTIC(O))
+					continue
+				if(prob(stage*stage))
+					O.damage = min(O.damage+stage, O.max_damage)
+
+	if(stage >= 4) //after 15 minutes
+		if(prob(3))
+			owner.reagents.add_reagent(egg, 2) //malignant cells breaking off, enter circulatory/lymphatic nodes to spread
+
+/mob/living/carbon/human/proc/infest_with_parasite(var/mob/living/carbon/victim, var/parasite_type, var/obj/item/organ/external/organ_to_infest, var/chance_to_infest = 100, var/parasite_limit = 3)
+	if(ishuman(victim))
+		var/mob/living/carbon/human/H = victim
+		if(BP_IS_ROBOTIC(organ_to_infest))
+			return
+		var/i = 0
+		for(var/obj/item/organ/internal/O in H.internal_organs) //dont want there to be a silly amount of parasites
+			if(istype(O, /obj/item/organ/internal/parasite))
+				i++
+		if((i < parasite_limit) && prob(chance_to_infest))
+			switch(parasite_type) //such a shitty way to do this but i couldnt get a better alternative to work in a sane amount of time :/
+				if("malignant tumour")
+					var/obj/item/organ/internal/parasite/malignant_tumour/P = new()
+					P.parent_organ = organ_to_infest
+					P.replaced(H, organ_to_infest)
+				if("benign tumour")
+					var/obj/item/organ/internal/parasite/benign_tumour/P = new()
+					P.parent_organ = organ_to_infest
+					P.replaced(H, organ_to_infest)
```

Comment:
drawing attention to this as it's not good lol.
i wasn't able to send a parasite's object path into the proc's var/obj/item/organ/internal/parasite/parasite_type parameter (before it was simplified to just receiving an organ_tag. might just be missing an obvious way to send an object path as a parameter and can easily de-shitcode this if pointed out, but what i had didn't work